README.md

```diff
@@ -1,16 +1,23 @@
-# Kubernetes Template Project
+# csi lib-iscsi
 
-The Kubernetes Template Project is a template for starting new projects in the GitHub organizations owned by Kubernetes. All Kubernetes projects, at minimum, must have the following files:
+A simple go package intended to assist CSI plugin authors by providing a tool set to manage iscsi connections.
 
-- a `README.md` outlining the project goals, sponsoring sig, and community contact information
-- an `OWNERS` with the project leads listed as approvers ([docs on `OWNERS` files][owners])
-- a `CONTRIBUTING.md` outlining how to contribute to the project
-- an unmodified copy of `code-of-conduct.md` from this repo, which outlines community behavior and the consequences of breaking the code
-- a `LICENSE` which must be Apache 2.0 for code projects, or [Creative Commons 4.0] for documentation repositories, without any custom content
-- a `SECURITY_CONTACTS` with the contact points for the Product Security Team 
-  to reach out to for triaging and handling of incoming issues. They must agree to abide by the
-  [Embargo Policy](https://github.com/kubernetes/sig-release/blob/master/security-release-process-documentation/security-release-process.md#embargo-policy) 
-  and will be removed and replaced if they violate that agreement.
+## Goals
+
+Provide a basic, lightweight library for CSI Plugin Authors to leverage some of the common tasks like connecting
+and disconnecting iscsi devices to a node.
+
+We inentionally avoid pulling in additional dependencies, and we intend to be stateless
+and as such are not using receivers.  Currently the focus is strictly based on a CSI context.
+
+## Design Philosophy
+
+The idea is to keep this as lightweight and generic as possible.  We intentionally avoid the use of any third party
+libraries or packages in this project.  We don't have a vendor directory, because we attempt to rely only on the std
+golang libs.  This may prove to not be ideal, and may be changed over time, but initially it's a worthwhile goal.
+
+Note that we're not using glog, or logrus, that's sure to raise hackles for folks.  We probably don't need them, and
+we may or may not even need logging in the lib at all.  For those that require logrus or glog however the simple
```

Comment:
This is missing the rest of the explanation for how to use logrus or glog with this lib

Reply:
Yeah, I wrote that before changing some things, I guess we should decide on a couple things:
1. Do we need logging in the lib (I think yes)
2. Should we ignore any fancy level setting and just provide simple logging using built in log
3. Use glog (one problem is that everybody seems to have a different preference for their plugin, so I'm inclined to just use something simple/built in)